node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs

```diff
@@ -0,0 +1,314 @@
+        let bodies: Vec<Value> = requests
+            .into_iter()
+            .map(|request| serde_json::from_slice(&request.body()).unwrap())
+            .collect();
+        assert_eq!(bodies[0]["method"].to_string(), "\"eth_getBalance\"",);
+        assert_eq!(
+            bodies[0]["params"][0].to_string(),
+            "\"0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc\"",
+        );
+        assert_eq!(bodies.len(), 1)
+    }
+
+    #[test]
+    #[should_panic(expected = "No address for an uninitialized wallet!")]
+    fn get_transaction_fee_balance_returns_err_for_an_invalid_wallet() {
+        let port = 8545;
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let chain = TEST_DEFAULT_CHAIN;
+        let subject = make_subject(transport, chain);
+
+        let result = subject.get_transaction_fee_balance(&Wallet::new("0x_invalid_wallet_address"));
+
+        assert_eq!(result, Err(BlockchainError::InvalidAddress));
+    }
+
+    #[test]
+    fn get_transaction_fee_balance_returns_err_for_unintelligible_response() {
+        let act = |subject: &RPCHelpersWeb3<Http>, wallet: &Wallet| {
+            subject.get_transaction_fee_balance(wallet)
+        };
+
+        assert_error_from_unintelligible_response(act, "invalid hex character");
+    }
+
+    #[test]
+    fn get_masq_balance_works() {
+        let port = find_free_port();
+        let test_server = TestServer::start (port, vec![
+            br#"{"jsonrpc":"2.0","id":0,"result":"0x00000000000000000000000000000000000000000000000000000000DEADBEEF"}"#.to_vec()
+        ]);
+        let (_event_loop_handle, transport) = Http::with_max_parallel(
+            &format!("http://{}:{}", &Ipv4Addr::LOCALHOST.to_string(), port),
+            REQUESTS_IN_PARALLEL,
+        )
+        .unwrap();
+        let chain = TEST_DEFAULT_CHAIN;
+        let subject = make_subject(transport, chain);
+
+        let result = subject
+            .get_masq_balance(
+                &Wallet::from_str("0x3f69f9efd4f2592fd70be8c32ecd9dce71c472fc").unwrap(),
+            )
+            .unwrap();
+
+        assert_eq!(result, U256::from(0xDEADBEEF_u64));
+        let requests = test_server.requests_so_far();
+        let bodies: Vec<Value> = requests
+            .into_iter()
+            .map(|request| serde_json::from_slice(&request.body()).unwrap())
+            .collect();
+        assert_eq!(bodies[0]["method"].to_string(), "\"eth_call\"",);
+        let contract_address = chain.rec().contract;
+        assert_eq!(
+            bodies[0]["params"][0]["to"].to_string(),
+            format!("\"{:?}\"", contract_address),
+        );
+        assert_eq!(bodies.len(), 1)
+    }
+
+    #[test]
+    #[should_panic(expected = "No address for an uninitialized wallet!")]
+    fn get_masq_balance_returns_err_for_an_invalid_wallet() {
```

Comment:
I think a lot of these tests are an overhead. We're testing the `web3` library over here, instead of some new logic that we've written.

Reply:
This file is gone now. 

But the truth is this test the blockchain interface using web3 code inside - an argument could be made that some of the methods of the trait of this interface are almost pure wrappers but still there isn't anything fundamentally wrong about testing wrappers, bear in mind that this is the prod code while probably vast majority of the higher level code uses the mock version. Then I ask who would test what is inside the real implication? What if there is mud instead of code?